numpy/_core/src/common/simd/lsx/operators.h

```diff
@@ -0,0 +1,301 @@
+#ifndef NPY_SIMD
+    #error "Not a standalone header"
+#endif
+
+#ifndef _NPY_SIMD_LSX_OPERATORS_H
+#define _NPY_SIMD_LSX_OPERATORS_H
+
+/***************************
+ * Shifting
+ ***************************/
+
+// left
+#define npyv_shl_u16(A, C) __lsx_vsll_h(A, npyv_setall_s16(C))
+#define npyv_shl_s16(A, C) __lsx_vsll_h(A, npyv_setall_s16(C))
+#define npyv_shl_u32(A, C) __lsx_vsll_w(A, npyv_setall_s32(C))
+#define npyv_shl_s32(A, C) __lsx_vsll_w(A, npyv_setall_s32(C))
+#define npyv_shl_u64(A, C) __lsx_vsll_d(A, npyv_setall_s64(C))
+#define npyv_shl_s64(A, C) __lsx_vsll_d(A, npyv_setall_s64(C))
+
+// left by an immediate constant
+#define npyv_shli_u16 __lsx_vslli_h
+#define npyv_shli_s16 __lsx_vslli_h
+#define npyv_shli_u32 __lsx_vslli_w
+#define npyv_shli_s32 __lsx_vslli_w
+#define npyv_shli_u64 __lsx_vslli_d
+#define npyv_shli_s64 __lsx_vslli_d
+
+// right
+#define npyv_shr_u16(A, C) __lsx_vsrl_h(A, npyv_setall_u16(C))
+#define npyv_shr_s16(A, C) __lsx_vsra_h(A, npyv_setall_u16(C))
+#define npyv_shr_u32(A, C) __lsx_vsrl_w(A, npyv_setall_u32(C))
+#define npyv_shr_s32(A, C) __lsx_vsra_w(A, npyv_setall_u32(C))
+#define npyv_shr_u64(A, C) __lsx_vsrl_d(A, npyv_setall_u64(C))
+#define npyv_shr_s64(A, C) __lsx_vsra_d(A, npyv_setall_u64(C))
+
+// Right by an immediate constant
+NPY_FINLINE __m128i npyv_shri_u16(__m128i a, unsigned int imm)
+{
+    if (imm > 15)
```

Comment:
It's not needed to do this check ,when imm bigger than 15 , instruction 'vsrl' still can get right result 0.
check the following func.

Reply:
Instruction not supported, compiler error Immediate overflow

Reply:
vsrl.h instruction need ui4, it will overflow if imm>15. Other functions are similar.